### ISSUE TITLE
Unable to create an upcloud_managed_object_storage resource with a public network

### ISSUE THREAD

**Issue by OlleLarsson** [UpCloudLtd/terraform-provider-upcloud#425]
Hi, as the title suggests, I'm not able to create a managed object storage resource with a public network specification

```
$ terraform version
Terraform v1.3.9
on linux_amd64
+ provider registry.terraform.io/upcloudltd/upcloud v3.0.1
```

Below is the terraform code that I'm trying to execute
```
terraform {
  required_providers {
    upcloud = {
      source  = "UpCloudLtd/upcloud"
      version = "~> 3.0"
    }
  }
}

provider "upcloud" {
  username = var.UPCLOUD_USERNAME
  password = var.UPCLOUD_PASSWORD
}

resource "upcloud_managed_object_storage" "managed_object_storage" {
  region            = "europe-1"
  configured_status = "started"

  network {
    family = "IPv4"
    name   = "pub"
    type   = "public"
  }

  users = ["olle_larsson"]

  labels = {
    managed-by = "terraform"
  }
}
```

Executing it gives me the following error
```
│ Error: error: message="Validation error.", type="https://developers.upcloud.com/1.3/errors#ERROR_INVALID_REQUEST", correlation_id=01HDGTKEWBMZW8YR51CCFRNWCQ, invalid_params_networks.0.uuid=''' is not valid 'uuid'.'
│ 
│   with upcloud_managed_object_storage.managed_object_storage,
│   on main.tf line 15, in resource "upcloud_managed_object_storage" "managed_object_storage":
│   15: resource "upcloud_managed_object_storage" "managed_object_storage" {
```

Setting a random uuid in the network specification then of course [yields the opposite error](https://github.com/UpCloudLtd/terraform-provider-upcloud/blob/v3.0.0/internal/service/managedobjectstorage/managed_object_storage.go#L295-L297), that uuid cannot be part of a public network specification.

Something obvious I'm missing or doing wrong :) ?

**Comment by villevsv-upcloud** [comment]
Thank you for reporting the issue.

We have identified the bug in network validation and a fix is on the way. I'll post an update & close the issue once a new release is available in the Terraform Registry.

**Comment by villevsv-upcloud** [comment]
A fix has been issued with the latest [v3.0.2 release](https://registry.terraform.io/providers/UpCloudLtd/upcloud/3.0.2).

I'll close the issue as solved.
